src/db_parser.jl

```diff
@@ -63,545 +353,1075 @@ function map_prime_mover(prime_mover::String)
+            region=collect(
+                IS.get_components(x -> get_id(x) == area, RegionTopology, p.data),
+            ),
+            value_of_lost_load=1e8, #TODO: Assign a default value to this field
+            power_systems_type=string(component_type),
         )
-        add_component!(system, b)
-        push!(buses, b)
+        add_technology!(p, d)
     end
+end
 
-    #Transmission Lines
-    lines = df_dict["transmission_lines"]
-    for row in eachrow(lines)
-        arc = PSY.Arc(;
-            from=filter(b -> contains(b.name, row["balancing_topology_from"]), buses)[1],
-            to=filter(b -> contains(b.name, row["balancing_topology_to"]), buses)[1],
+function add_loads!(p::Portfolio, attributes::Dict{Int64, Dict{String, Any}}, db::SQLite.DB)
+    # stream straight through the table
+    for rec in DBInterface.execute(db, QUERIES[:demand_requirements])
+        component_type = eval(
+            Meta.parse(first(DBInterface.execute(db, QUERIES[:entity_type], [rec.id]))[1]),
         )
-
-        tx = Line(;
-            name=string("line_", rownumber(row)),
-            available=true,
-            active_power_flow=0.0,
-            reactive_power_flow=0.0,
-            arc=arc,
-            r=1.0,
-            x=1.0,
-            b=(from=1.0, to=1.0),
-            rating=row["continuous_rating"],
-            angle_limits=(-1.5, 1.5),
+        component_attr = get(attributes, rec.id, Dict{String, Any}())
+
+        # Determine area based on balancing topology if zonal
+        bus_name = first(
+            DBInterface.execute(db, QUERIES[:topology_for_unit], [rec.balancing_topology]),
+        )[1]
+
+        # build and immediately insert
+        d = component_type(;
+            name=rec.name,
+            base_power=rec.base_power,
+            bus=PSY.get_component(ACBus, p.base_system, bus_name),
+            max_active_power=component_attr["active_power"] / rec.base_power, #TODO: Change to "max_active_power" later when DB is fixed
+            max_reactive_power=component_attr["max_reactive_power"] / rec.base_power,
+            reactive_power=component_attr["reactive_power"] / rec.base_power,
+            active_power=component_attr["active_power"] / rec.base_power,
+            available=component_attr["available"],
         )
-        add_component!(system, tx)
+        if haskey(component_attr, "uuid")
+            IS.set_uuid!(IS.get_internal(d), Base.UUID(component_attr["uuid"]))
+        end
+        add_component!(p.base_system, d)
     end
+end
 
-    # Get existing generation units
-    for row in eachrow(df_dict["generation_units"])
+function transform_natural_impedance_to_device_base(natural_units_impedance, arc, sys)
+    base_voltage = arc.from.base_voltage
+    if isnothing(base_voltage)
+        error("Base voltage is not defined")
+    end
+    base_power = get_base_power(sys)
+    Z_base = base_voltage^2 / base_power
```

Comment:
```suggestion
    z_base = base_voltage^2 / base_power
```
Always use lower case so that developers don't have to guess. We don't need to preserve everything about the math.